internal/local/app.go

```diff
@@ -265,26 +306,33 @@ func FindApp(path string) (App, bool, error) {
 	}
 
 	for i := 0; i < maxDirectoryContainSearchDepth; i++ {
```

Comment:
(haven't thoroughly tested but) what do you guys think about something more like this?
```
func resolveConfig(path string) (App, bool, error) {
	app := App{RootDir: path}
	if _, err := os.Stat(filepath.Join(path, FileMdbAppMeta.String())); err == nil {
		app.LoadAppMeta()
		switch app.AppMeta.ConfigVersion {
		case realm.AppConfigVersion20180301:
			app.Config = FileStitch
		case realm.AppConfigVersion20200603:
			app.Config = FileConfig
		case realm.AppConfigVersion20210101:
			app.Config = FileRealmConfig
		}
		return app, true, nil
	}

	for _, config := range allConfigFiles {
		_, err := os.Stat(filepath.Join(path, config.String()))
		if err != nil {
			if os.IsNotExist(err) {
				continue
			}
			return App{}, false, err
		}
		app.Config = config
		return app, true, nil
	}

	return App{}, false, nil
}

// FindApp searches upwards for the root of a Realm app project and
// returns the local app structure, a boolean indicating if the current
// working directory is part of a Realm app project, and any error that occurs
func FindApp(path string) (App, bool, error) {
	wd, wdErr := filepath.Abs(path)
	if wdErr != nil {
		return App{}, false, wdErr
	}

	for i := 0; i < maxDirectoryContainSearchDepth; i++ {
		app, found, err := resolveConfig(wd)
		if err != nil {
			return App{}, false, err
		}
		if !found {
			continue
		}
		if err := app.LoadConfig(); err != nil {
			return App{}, false, err
		}

		// if no config version is found then continue searching for the app's root directory config
		if app.ConfigVersion() == realm.AppConfigVersionZero {
			continue
		}
		return app, true, nil
		if wd == "/" {
			break
		}
		wd = filepath.Clean(filepath.Join(wd, ".."))
	}

	return App{}, false, nil
}
```

Reply:
I like this approach of resolving the config. There's some error handling edge cases I think you're missing but I'll work on this approach.